examples/tests/sdk_tests/std/map.test.w

```diff
@@ -15,3 +15,210 @@ let mvalues = m.values();
+}
+
+test "keys()" {
+    let lightsaberColorMap = { "red" => "sith", "blue" => "jedi" };
+    let lightsaberColors = lightsaberColorMap.keys();
+    let colorArr = ["red", "blue"];
+    assert(lightsaberColors.length == colorArr.length);
+    for i in 0..lightsaberColors.length {
+        assert(lightsaberColors.at(i) == colorArr.at(i));
+    }
+
+    let isMandalorianWarrior = MutMap<bool> { "bo katan" => true, "jango" => true, "satine" => false, "boba" => true};
+    let mandalorianKeys = isMandalorianWarrior.keys();
+    let keysArr = ["bo katan", "jango", "satine", "boba"];
+    assert(mandalorianKeys.length == keysArr.length);
+    for i in 0..mandalorianKeys.length {
+        assert(mandalorianKeys.at(i) == keysArr.at(i));
+    }
+    
+}
+
+
+let ruleOfTwo = { "master" => "apprentice" };
+let mutRuleOfTwo: MutMap<str> = ruleOfTwo.copyMut();
+assert(mutRuleOfTwo.copy() == ruleOfTwo);
+assert(mutRuleOfTwo.delete("master") == true);
+assert(mutRuleOfTwo.size() == 0);
+
+test "copyMut()" {
+    let ruleOfTwo = { "master" => "apprentice" };
+    let mutRuleOfTwo: MutMap<str> = ruleOfTwo.copyMut();
+    assert(mutRuleOfTwo.copy() == ruleOfTwo);
+    assert(mutRuleOfTwo.delete("master") == true);
+    assert(mutRuleOfTwo.size() == 0);
+}
+
+
+let authority = MutMap<str> {"republic" => "senate", "empire" => "emperor"};
+let immutAuthority: Map<str> = authority.copy();
+assert(immutAuthority.copyMut() == authority);
+
+test "copy()" {
+    let authority = MutMap<str> {"republic" => "senate", "empire" => "emperor"};
+    let immutAuthority: Map<str> = authority.copy();
+    assert(immutAuthority.copyMut() == authority);
+}
+
+
+let senate = MutMap<str> {"chancellor" => "palpatine"}; // I love democracy
+senate.set("senator", "organa");
+assert(senate.get("senator") == "organa");
+assert(senate.size() == 2);
+
+test "set()" {
+    let senate = MutMap<str> {"chancellor" => "palpatine"};
+    senate.set("senator", "organa");
+    assert(senate.get("senator") == "organa");
+    assert(senate.size() == 2);
+}
+
+
+let position = MutMap<str> {"librarian" => "jocasta"};
+position.clear();
+assert(position.size() == 0);
+
+test "clear()" {
+    let position = MutMap<str> {"librarian" => "jocasta"};
+    position.clear();
+    assert(position.size() == 0);
+}
+
+
+let sithTriumvirate = MutMap<str> { "traya" => "lord of betrayal", "nihilus" => "lord of hunger", "sion" => "lord of pain" };
+sithTriumvirate.delete("nihilus");
+assert(sithTriumvirate.size() == 2);
+sithTriumvirate.delete("sion");
+assert(sithTriumvirate == MutMap<str> { "traya" => "lord of betrayal" });
+
+test "delete()" {
+    let sithTriumvirate = MutMap<str> { "traya" => "lord of betrayal", "nihilus" => "lord of hunger", "sion" => "lord of pain" };
+    sithTriumvirate.delete("nihilus");
+    assert(sithTriumvirate.size() == 2);
+    sithTriumvirate.delete("sion");
+    assert(sithTriumvirate == MutMap<str> { "traya" => "lord of betrayal" });
+}
```

Comment:
I'd add a check for testing if calling `set`, `delete`, `clear` and `copy` on a non Mut Map throws an error too

Reply:
I actually tried one, I think. But the try-catch wasn't able to catch that error and it was instead failing. It was something like;
```
let a = MutMap<str> {};
let b = a.copy();
try {
    a.delete()
} catch error {
    ## not sure if catch error or just catch is the correct syntax but both fails
    throw("")
}
```

Now this, instead of catching the error, would result in a compilation error inside the try block. So I couldn't figure out how to do it. Any suggestions?

Reply:
Right! 
that's why we got the invalid tests (tests in the `examples/tests/invalid` folder, that catch compilation errors)
I see that there is only one test for `set` there, can you add tests for the rest?  (it's in examples/tests/invalid/immutable_container_types.test.w, maybe we can change the name of the file)